app/models/user.rb

```diff
@@ -0,0 +1,38 @@
+class User < ApplicationRecord
+  has_secure_password
+
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  EMAIL_MIN_LENGTH = 20
+  EMAIL_MAX_LENGTH = 255
+  NAME_MAX_LENGTH = 10
+  PASSWORD_MIN_LENGTH = 6
+  MAX_BIRTHDAY_RANGE = 100
+
+  before_save :downcase_email
+
+  validates :email,
+            presence: true,
+            length: {minimum: EMAIL_MIN_LENGTH, maximum: EMAIL_MAX_LENGTH},
+            format: {with: VALID_EMAIL_REGEX},
+            uniqueness: {case_sensitive: false}
+
+  validates :name, presence: true, length: {maximum: NAME_MAX_LENGTH}
+
+  validates :password, presence: true, length: {minimum: PASSWORD_MIN_LENGTH}
```

Comment:
```suggestion
  validates :email,
            presence: true,
            length: {minimum: EMAIL_MIN_LENGTH, maximum: EMAIL_MAX_LENGTH},
            format: {with: VALID_EMAIL_REGEX},
            uniqueness: {case_sensitive: false}
  validates :name, presence: true, length: {maximum: NAME_MAX_LENGTH}
  validates :password, presence: true, length: {minimum: PASSWORD_MIN_LENGTH}
```
Các validates này cùng loại nên k cần line trống